scripts/setup-seafile-mysql.py

```diff
@@ -1227,11 +1231,129 @@ def set_file_perm():
+
+    seafile_dir = get_param_val(args.seafile_dir, 'SEAFILE_DIR',
+                                os.path.join(env_mgr.top_dir, 'seafile-data'))
+    seafile_config.seafile_dir = seafile_config.validate_seafile_dir(seafile_dir)
+
+    global db_config
+
+    use_existing_db = get_param_val(args.use_existing_db, 'USE_EXISTING_DB', '1')
+    if use_existing_db == '1':
+        db_config = NewDBConfigurator()
+    elif use_existing_db == '2':
+        db_config = ExistingDBConfigurator()
+    else:
+        raise InvalidParams('Invalid use-existing-db parameter, the value can only be 1 or 2')
+
+    mysql_host = get_param_val(args.mysql_host, 'MYSQL_HOST', '127.0.0.1')
+    if not mysql_host:
+        raise InvalidParams('Incomplete mysql parameters, miss mysql host')
+    db_config.mysql_host = db_config.validate_mysql_host(mysql_host)
+
+    mysql_port = get_param_val(args.mysql_port, 'MYSQL_PORT', '3306')
+    db_config.mysql_port = Utils.validate_port(mysql_port)
+
+    mysql_user = get_param_val(args.mysql_user, 'MYSQL_USER')
+    if not mysql_user:
+        raise InvalidParams('Incomplete mysql parameters, miss mysql user')
+
+    mysql_user_passwd = get_param_val(args.mysql_user_passwd, 'MYSQL_USER_PASSWD')
+    if not mysql_user_passwd:
+        raise InvalidParams('Incomplete mysql parameters, miss mysql user passwd')
+
+    ccnet_db = get_param_val(args.ccnet_db, 'CCNET_DB', 'ccnet-db')
+    if not ccnet_db:
+        raise InvalidParams('Incomplete mysql parameters, miss ccnet db')
+
+    seafile_db = get_param_val(args.seafile_db, 'SEAFILE_DB', 'seafile-db')
+    if not seafile_db:
+        raise InvalidParams('Incomplete mysql parameters, miss seafile db')
+
+    seahub_db = get_param_val(args.seahub_db, 'SEAHUB_DB', 'seahub-db')
+    if not seahub_db:
+        raise InvalidParams('Incomplete mysql parameters, miss seahub db')
+
+    mysql_user_host = get_param_val(args.mysql_user_host, 'MYSQL_USER_HOST')
+    mysql_root_passwd = get_param_val(args.mysql_root_passwd, 'MYSQL_ROOT_PASSWD')
+
+    if db_config.use_existing_db:
+        db_config.check_mysql_user(mysql_user, mysql_user_passwd)
+        db_config.seafile_mysql_user = mysql_user
+        db_config.seafile_mysql_password = mysql_user_passwd
+        db_config.ccnet_db_name = db_config.validate_db_name(ccnet_db)
+        db_config.seafile_db_name = db_config.validate_db_name(seafile_db)
+        db_config.seahub_db_name = db_config.validate_db_name(seahub_db)
+    else:
+        if db_config.mysql_host != '127.0.0.1' and not mysql_user_host:
+            raise InvalidParams('In create db mode, miss mysql user host parameter')
+        if not mysql_user_host:
+            db_config.mysql_userhost = 'localhost'
+        else:
+            db_config.mysql_userhost = db_config.validate_mysql_user_host(mysql_user_host)
+
+        if not mysql_root_passwd:
+            raise InvalidParams('In create db mode, miss mysql root passwd parameter')
```

Comment:
Pay attention to the syntax and readability of all messages printed to the user.